src/OrchardCore/OrchardCore.Data.YesSql.Abstractions/DbConnectionValidatorResult.cs

```diff
@@ -2,21 +2,39 @@ namespace OrchardCore.Data;
 
 public enum DbConnectionValidatorResult
 {
-    // Unknown indicates that the connection string status is unknown or was not yet validated
+    /// <summary>
+    /// 'Unknown' indicates that the connection string status is unknown or was not validated.
+    /// </summary>
     Unknown,
 
-    // NoProvider indicated that the provider is missing
+    /// <summary>
+    /// 'NoProvider' indicated that the database provider is missing.
+    /// </summary>
     NoProvider,
 
-    // DocumentNotFound indicates that the connection string was valid, yet the Document table does not exist 
-    DocumentNotFound,
+    /// <summary>
+    /// 'DocumentTableNotFound' indicates that the connection string was valid, while the Document table does not exists.
+    /// </summary>
+    DocumentTableNotFound,
 
-    // DocumentFound indicates that the connection string was valid, yet the Document table exist
-    DocumentFound,
+    /// <summary>
+    /// 'DocumentTableFound' indicates that the connection string was valid, while the Document table exists.
+    /// </summary>
+    DocumentTableFound,
 
-    // InvalidConnection unable to open a connection to the given connection string
+    /// <summary>
+    /// 'ShellDescriptorDocumentFound' indicates that the connection string was valid and the document table exists with at least one shell-descriptor document.
+    /// This also means that the Document table is used by a tenant.
+    /// </summary>
+    ShellDescriptorDocumentFound,
```

Comment:
@jtkech are you agree to introduce a new database connection validation status related to ShellDescriptor?

Reply:
@hishamco Yes, in case of the Default tenant we allow that the Document table is already used by an host level service, as it happens e.g. when shell settings come from the database in place of `tenants.json`, this by using from the app startup `.AddDatabaseShellsConfiguration()`.

So in that case we allow that the Document table already exists but it should not contain the **ShellDescriptor** document meaning that an initialized tenant already uses it.

We could have done the same check for any tenants by allowing any tenant to use the same table prefix than e.g. the shells settings from database, but in the future we will be able to remove any non Default tenant and then its Document table, so we may lose e.g. all tenant settings.

Fortunally we can't remove the Default tenant.